python/cudf/cudf/_cuda/gpu.pyx

```diff
@@ -307,8 +320,32 @@ def getDeviceCount():
 
     cdef int count
     cdef cudaError_t status = cudaGetDeviceCount(&count)
+
     if status != 0:
-        raise CUDARuntimeError(status)
+        notify_caller_errors = set([
+            cudaErrorInitializationError,
+            cudaErrorInsufficientDriver,
+            cudaErrorInvalidDeviceFunction,
+            cudaErrorNoDevice,
+            cudaErrorInvalidDevice,
+            cudaErrorStartupFailure,
+            cudaErrorInvalidKernelImage,
+            cudaErrorDeviceUninitialized,
+            cudaErrorAlreadyAcquired,
+            cudaErrorOperatingSystem,
+            cudaErrorNotPermitted,
+            cudaErrorNotSupported,
+            cudaErrorSystemNotReady,
+            cudaErrorSystemDriverMismatch,
+            cudaErrorCompatNotSupportedOnDevice,
+            cudaErrorTimeout,
+            cudaErrorUnknown,
+            cudaErrorApiFailureBase
+        ])
```

Comment:
```suggestion
        notify_caller_errors = {
            cudaErrorInitializationError,
            cudaErrorInsufficientDriver,
            cudaErrorInvalidDeviceFunction,
            cudaErrorNoDevice,
            cudaErrorInvalidDevice,
            cudaErrorStartupFailure,
            cudaErrorInvalidKernelImage,
            cudaErrorDeviceUninitialized,
            cudaErrorAlreadyAcquired,
            cudaErrorOperatingSystem,
            cudaErrorNotPermitted,
            cudaErrorNotSupported,
            cudaErrorSystemNotReady,
            cudaErrorSystemDriverMismatch,
            cudaErrorCompatNotSupportedOnDevice,
            cudaErrorTimeout,
            cudaErrorUnknown,
            cudaErrorApiFailureBase
        }
```

Reply:
We should probably also define this outside the function so as to not need to recreate it on each call.